tools/proofers/processtext.php

```diff
@@ -347,7 +347,7 @@
 function switch_layout()
 {
     $user = User::load_current();
-    $user->profile->i_layout = $user->profile->i_layout == 1 ? 0 : 1;
+    $user->profile->i_layout = $user->profile->i_layout == true ? false : true;
```

Comment:
This is a good find, but I think the error is in the UserProfile type specification for this, I think the following types are wrong:
```
 * @property bool $i_res;  # should be an int, values are in pinc/resolution.inc
 * @property bool $i_type;  # should be an int -- yes the values are just 0 and 1 but that's really a 2-value enum
 * @property bool $i_layout;  # same about a 2-value enum
 
 * @property bool $v_fntf;  # should be an int -- multi-valued enum
 * @property bool $h_fntf;  # same as ^^

```

Reply:
Sounds good, I have applied the fix to the suggested types and reverted the int-to-bool change.